### PR TITLE
[KernelGen][Nvidia] Add trunc_ operator with Triton kernel

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
Adds a Triton kernel for `trunc_`. Rounds each element of the input tensor to the nearest integer towards zero.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_trunc_.py --level core` (NVIDIA CI)

### trunc

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|-------|------|--------------------|-------------------|---------|--------|
| float16 | [1073741824] | 3.109888 | 3.600384 | 0.864 | 0.298 |
| float16 | [64, 64] | 0.007168 | 0.007168 | 1.000 | 0.001 |
| float16 | [4096, 4096] | 0.058368 | 0.059392 | 0.983 | 0.282 |
| float16 | [64, 512, 512] | 0.057344 | 0.060416 | 0.949 | 0.278 |
| float16 | [1024, 1024, 1024] | 3.107840 | 3.655680 | 0.850 | 0.294 |
| float32 | [1073741824] | 6.214656 | 6.377472 | 0.974 | 0.168 |
| float32 | [64, 64] | 0.008192 | 0.008192 | 1.000 | 0.001 |
| float32 | [4096, 4096] | 0.106496 | 0.106496 | 1.000 | 0.158 |
| float32 | [64, 512, 512] | 0.106496 | 0.105472 | 1.010 | 0.159 |
| float32 | [1024, 1024, 1024] | 6.215680 | 6.397952 | 0.972 | 0.168 |
| bfloat16 | [1073741824] | 3.108864 | 3.483648 | 0.892 | 0.308 |
| bfloat16 | [64, 64] | 0.008192 | 0.007168 | 1.143 | 0.001 |
| bfloat16 | [4096, 4096] | 0.057344 | 0.059392 | 0.966 | 0.282 |
| bfloat16 | [64, 512, 512] | 0.058368 | 0.060416 | 0.966 | 0.278 |
| bfloat16 | [1024, 1024, 1024] | 3.108864 | 3.662848 | 0.849 | 0.293 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| trunc | **0.961** |

### trunc_ (in-place)

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|-------|------|--------------------|-------------------|---------|--------|
| float16 | [1073741824] | 3.131392 | 3.650560 | 0.858 | 0.294 |
| float16 | [64, 64] | 0.007168 | 0.007168 | 1.000 | 0.001 |
| float16 | [4096, 4096] | 0.057344 | 0.056320 | 1.018 | 0.298 |
| float16 | [64, 512, 512] | 0.056320 | 0.056320 | 1.000 | 0.298 |
| float16 | [1024, 1024, 1024] | 3.131392 | 3.650560 | 0.858 | 0.294 |
| float32 | [1073741824] | 6.336512 | 6.513664 | 0.973 | 0.165 |
| float32 | [64, 64] | 0.007168 | 0.007168 | 1.000 | 0.001 |
| float32 | [4096, 4096] | 0.106496 | 0.104448 | 1.020 | 0.161 |
| float32 | [64, 512, 512] | 0.106496 | 0.103424 | 1.030 | 0.162 |
| float32 | [1024, 1024, 1024] | 6.337536 | 6.509568 | 0.974 | 0.165 |
| bfloat16 | [1073741824] | 3.131904 | 3.657728 | 0.856 | 0.294 |
| bfloat16 | [64, 64] | 0.007168 | 0.007168 | 1.000 | 0.001 |
| bfloat16 | [4096, 4096] | 0.057344 | 0.057344 | 1.000 | 0.293 |
| bfloat16 | [64, 512, 512] | 0.056320 | 0.056320 | 1.000 | 0.298 |
| bfloat16 | [1024, 1024, 1024] | 3.131392 | 3.656704 | 0.856 | 0.294 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| trunc_ (in-place) | **0.963** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (CI) | PASS (30 cases) | 0.961/0.963 | Primary |
| Tianshu | PASS | 1.025 | — |
| Muxi | PASS | 1.006 | — |
| Ascend | PASS | — | — |
| Hygon | PASS | 1.036 | — |

## Files Changed
- `src/flag_gems/ops/trunc_.py`: Triton kernel implementation
- `tests/test_trunc_.py`: Accuracy test
- `benchmark/test_trunc_.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Math, stage: alpha 5.1)

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -6390,6 +6390,28 @@ ops:
...
+    for:
+      - trunc_
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
+      - alpha: '5.1'
+  - id: trunc__
```
Comment (Dongxu-H) [inline conf/operators.yaml]: trunc and trunc_
not
trunc_ and trunc__

File: benchmark/test_trunc_.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import torch
+
+from . import base, consts
+
+
+@pytest.mark.trunc_
+def test_trunc_():
```
Comment (Dongxu-H) [inline benchmark/test_trunc_.py:18]: two kernels, one test

File: conf/operators.yaml (hunk 1 above)
Comment (XDYuanzhuLee) [reply conf/operators.yaml]: Fixed: renamed yaml entries to  and , matching repo convention (e.g. abs/abs_). Also fixed function names in the implementation file and updated imports/registration.

File: benchmark/test_trunc_.py (hunk 2 above)
Comment (XDYuanzhuLee) [reply benchmark/test_trunc_.py:18]: Fixed: added  benchmark for the out-of-place kernel alongside the existing . Both kernels now have separate benchmark coverage.

File: conf/operators.yaml (hunk 1 above)
Comment (XDYuanzhuLee) [reply conf/operators.yaml]: Fixed: renamed yaml entries to trunc and trunc_, matching repo convention (e.g. abs/abs_). Also fixed function names in the implementation file and updated imports/registration.

File: benchmark/test_trunc_.py (hunk 2 above)
Comment (XDYuanzhuLee) [reply benchmark/test_trunc_.py:18]: Fixed: added test_trunc benchmark for the out-of-place kernel alongside the existing test_trunc_. Both kernels now have separate benchmark coverage.

**Review by w1120029931-bit — APPROVED** [review]
